inductiva/mixins/file_manager.py

```diff
@@ -0,0 +1,230 @@
+        if self.__root_dir is None:
+            self.set_root_dir(self.DEFAULT_ROOT_DIR)
+
+    def set_root_dir(self, root_dir=None):
+        """Set a root directory for the file manager.
+        
+        All files managed through the manager will be
+        placed inside a newly created folder with the given
+        name. If a folder with the same name already exists,
+        the newly created one will be appended with the
+        "#1" suffix. If folders having the "#N" suffix already
+        exist, the given name will be appended with "#{N+1}"
+        where N is determined by the largest existing N.
+
+        Args:
+            root_dir: Path to the root directory.
+                If None, an error is raised.
+                If the directory already exists. A new directory
+                name is created following a sequence of #1, #2, #3, etc."""
+
+        if root_dir is None:
+            raise ValueError("Given root directory cannot be None")
+        elif os.path.isdir(root_dir):
+            if os.getenv("INDUCTIVA_DISABLE_FILEMANAGER_AUTOSUFFIX", False):  #pylint: disable=invalid-envvar-default
+                raise FileExistsError(f"Directory {root_dir} already exists.")
+            generated_root_dir = _gen_unique_name(root_dir)
+            logging.info(
+                "Directory %s already exists."
+                " Setting root folder to %s.", root_dir, generated_root_dir)
+            root_dir = generated_root_dir
+
+        root_dir = inductiva.utils.files.resolve_path(root_dir)
+        os.makedirs(root_dir)
+        self.__root_dir = root_dir
+
+    def get_root_dir(self):
+        """Get the active root directory for the file manager."""
+
+        self.__check_root_dir()
+
+        return self.__root_dir
+
+    def add_file(self, source_file, target_file=None, **render_args):
+        """Render a file from a template.
+        
+        Renders the given `source_file` using a templating mechanism and writes
+        the output to `target_file`.
+        If `target_file` is None, the output is saved inside the manager's root
+        directory to a file with the same name as the source file, but with the
+        template suffix removed.
+        Only template files with the template extension are rendered.
+        Other files are copied.
+
+        Args:
+            source_file: Path to the source file.
+            target_file: Path to the target file.
+            render_args: Arguments to render the template file.
+        """
+
+        self.__check_root_dir()
+
+        if target_file is None:
+            new_target_file = pathlib.Path(source_file).name
+        else:
+            new_target_file = target_file
+
+        new_target_file = os.path.join(self.__root_dir, new_target_file)
+        if source_file.endswith(TEMPLATE_EXTENSION):
+            if target_file is None:
+                new_target_file = new_target_file.split(TEMPLATE_EXTENSION)[0]
```

Comment:
will break. example:

```python
>>> TEMPLATE_EXTENSION='txt'
>>> new_target_file = 'a.b.c.txt.txt.txt'
>>> new_target_file.split(TEMPLATE_EXTENSION)[0]
'a.b.c'
```

use:
```suggestion
                new_target_file, _ = os.path.splitext(new_target_file)
```